Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource/helper.go

```diff
@@ -117,7 +117,7 @@ func (m *Helper) Create(namespace string, modify bool, data []byte) (runtime.Obj
 			if err := m.Versioner.SetResourceVersion(obj, ""); err != nil {
 				return nil, err
 			}
-			newData, err := m.Codec.Encode(obj)
+			newData, err := m.RESTClient.GetCodec().Encode(obj)
```

Comment:
That is the original pattern from the original implementation.  It's never
been important enough to change.

On Wed, Oct 28, 2015 at 10:46 AM, David Eads <notifications@github.com>
wrote:

> In Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource/helper.go
> <https://github.com/openshift/origin/pull/5454#discussion_r43264340>:
>
> > @@ -117,7 +117,7 @@ func (m *Helper) Create(namespace string, modify bool, data []byte) (runtime.Obj
> >  			if err := m.Versioner.SetResourceVersion(obj, ""); err != nil {
> >  				return nil, err
> >  			}
> > -			newData, err := m.Codec.Encode(obj)
> > +			newData, err := m.RESTClient.GetCodec().Encode(obj)
>
> I don't like the sig func (m *Helper) Create(namespace string, modify
> bool, data []byte) either, but changing it to func (m *Helper)
> Create(namespace string, modify bool, data interface{}) is more that I'd
> really like to change here
>
> Especially since someone (@smarterclayton
> <https://github.com/smarterclayton>) thought about this upstream and
> decided that passing through straight []byte was the proper way to build
> the API. They went to some effort to make that happen.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5454/files#r43264340>.
>